docs/transform.ipynb

```diff
@@ -75,102 +67,992 @@
+    "\n",
+    "\n",
+    "Cirq provides many out-of-the-box transformers which can be used as individual compilation passes and also provides a general framework for users to create their own transformers using powerful primitives and bundle a bunch of transformers together to enable compiling circuits for specific targets. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "KUor1pGZi0Iw"
+   },
+   "source": [
+    "# Built-in Transformers in Cirq"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "861ea1ada088"
+   },
+   "source": [
+    "## Overview\n",
+    "Transformers that come with cirq can be found in the `cirq.transformers` package.\n",
     "\n",
-    "Cirq comes with the concept of an optimizer.  Optimizers will pass over a circuit and perform tasks that will modify the circuit in place.  These can be used to transform a circuit in specific ways, such as combining single-qubit gates, commuting Z gates through the circuit, or readying the circuit for certain hardware or gate set configurations.\n",
+    "A few notable examples are:\n",
+    "*   **cirq.align_left / cirq.align_right**: Align gates to the left/right of the circuit.\n",
+    "*   **cirq.defer_measurements**:  Moves all (non-terminal) measurements in a circuit to the end of circuit by implementing the deferred measurement principle.\n",
+    "*   **cirq.drop_empty_moments** / **cirq.drop_negligible_operations**:  Removes moments that are empty or operations that have very small effects, respectively.\n",
+    "*   **cirq.eject_phased_paulis**: Pushes X, Y, and PhasedX gates towards the end of the circuit, potentially absorbing Z gates and modifying gates along the way.\n",
+    "*   **cirq.eject_z**:  Pushes Z gates towards the end of the circuit, potentially adjusting phases of gates that they pass through.\n",
+    "*   **cirq.expand_composite**:  Uses `cirq.decompose` to expand composite gates.\n",
+    "*   **cirq.merge_k_qubit_unitaries**: Replaces connected components of unitary operations, acting on <= k qubits, with op-tree given by `rewriter(circuit_op)`.\n",
+    "*   **cirq.optimize_for_target_gateset**: Attempts to convert a circuit into and equivalent circuit using only gates from a given target gateset.\n",
+    "*   **cirq.stratified_circuit**: Repacks the circuit to ensure that moments only contain operations from the same category.\n",
+    "*   **cirq.synchronize_terminal_measurements**:  Moves all terminal measurements in a circuit to the final moment, if possible.\n",
     "\n",
-    "Optimizers will have a function `optimize_circuit()` that can be used to perform this optimization.  Here is a simple example that removes empty moments:"
+    "\n",
+    "\n",
+    "Below we show a toy example of how to use these transformers to optimize a circuit. "
```

Comment:
Nit: we show. Second person you might want to do something like "Below, you can implement a transformer pipeline called `optimize_circuit`".